cmd/controlplane/controller.go

```diff
@@ -133,16 +132,15 @@ func newControllerCommand() *cobra.Command {
 				}
 			}
 
-			var argoClientForCreds client.Client
+			credentialsDbOpts := make([]credentials.KubernetesDatabaseOption, 0)
 			if types.MustParseBool(
 				os.GetEnv("ARGOCD_ENABLE_CREDENTIAL_BORROWING", "false"),
 			) {
-				argoClientForCreds = appMgr.GetClient()
+				credentialsDbOpts = append(credentialsDbOpts, credentials.WithArgoClient(appMgr.GetClient()))
```

Comment:
I don't think the client should be treated as an option, because it's 100% required. Could we have two args: client and opts?

Reply:
Sorry... I mis-spoke here.

The "Kargo client" is 100% required. This one is optional, depending on whether credential borrowing is enabled.

So... imho, treat the "Kargo client" as required and leave the "Argo client" in the realm of opts.

Reply:
In practice, I've seen a lot of "opts" usage for both required and "optional args".  

But makes sense if you wanna make this explicit